src/scripttabs.cpp

```diff
@@ -108,7 +108,7 @@ TYPE_SYMBOL asTypeTable[] =
 	{ "",	(INTERP_TYPE)ST_POINTER_STRUCTSTAT, AT_SIMPLE, NULL, NULL },	//for NULLSTRUCTURESTAT
 
 	/* This final entry marks the end of the type list */
-	{ "END OF TYPE LIST", (INTERP_TYPE)NULL,		AT_SIMPLE,	NULL,	NULL },
+	{ "END OF TYPE LIST", (INTERP_TYPE)0,		AT_SIMPLE,	NULL,	NULL },
```

Comment:
Having a lot of variables of `static` storage duration with complex constructors will significantly affect startup time (those are _all_ execute _before `main`). That's why I like to avoid things like that.

> Perhaps you've meant `end() == array + sizeof(array)/sizeof(array[0])` - yes, but with that you'll have to always pass and store length or end along with the array pointer.

Yes, you're correct, I almost got it right: luckily there's `std::begin` and `std::end` which provide the means of getting these for arrays without such easy to make accidents. I agree it still gives you the inconvenience of having to keep track of the size or end-marker somehow. Something like `iterator_range` (both from `ranges::v3` and `boost`) can make it a lot more convenient to do so than separate iterators.